Unexpected EOF after azurerm providers upgrade

### Terraform Version
Terraform v0.10.7

### AzureRM Provider Version
0.3.1

### Affected Resource(s)
Please list the resources as a list:
- azurerm_lb_rule

### Terraform Configuration Files
```hcl
resource "azurerm_lb_rule" "tr-coll-store-http" {
  resource_group_name            = "${data.azurerm_resource_group.tr.name}"
  loadbalancer_id                = "${azurerm_lb.tr-coll-store-lb.id}"
  name                           = "HTTPStoreRule"
  protocol                       = "Tcp"
  frontend_port                  = 80
  backend_port                   = 80
  frontend_ip_configuration_name = "tr-coll-lb-store-private"
  backend_address_pool_id        = "${azurerm_lb_backend_address_pool.tr-coll-store-bp.id}"
  probe_id                       = "${azurerm_lb_probe.tr-coll-store-lb-probe.id}"
}
```

### Debug Output
_azurerm_lb_rule.tr-coll-store-http: 1 error(s) occurred:
azurerm_lb_rule.tr-coll-store-http: azurerm_lb_rule.tr-coll-store-http: unexpected EOF_

### Actual Behavior
Terraform plan return an _azurerm_lb_rule.tr-coll-store-http: azurerm_lb_rule.tr-coll-store-http: unexpected EOF_

### Steps to Reproduce
1. `terraform init -upgrade`
2. `terraform plan`

I've tried with last provider version (0.3.2) and this is the output:

```
data.azurerm_resource_group.tr: Refreshing state...
azurerm_public_ip.yyyyyyyyyyy-ip: Refreshing state... (ID: /subscriptions/xxxxxxxxxxxxxxxxxxx-...ublicIPAddresses/xxxxxxxxxx-coll-llllllll)
azurerm_lb.yyyyyyyyyyy-lb: Refreshing state... (ID: /subscriptions/xxxxxxxxxxxxxxxxxxx-...Network/loadBalancers/yyyyyyyyyyy-lb)
azurerm_lb_backend_address_pool.yyyyyyyyyyy-bp: Refreshing state... (ID: /subscriptions/xxxxxxxxxxxxxxxxxxx-...ndAddressPools/BackEndAddressPoolllllllll)
azurerm_lb_probe.yyyyyyyyyyy-lb-probe: Refreshing state... (ID: /subscriptions/xxxxxxxxxxxxxxxxxxx-...oll-llllllll-lb/probes/http-running-probe)
azurerm_lb_nat_pool.yyyyyyyyyyy-lb-nat-pool: Refreshing state... (ID: /subscriptions/xxxxxxxxxxxxxxxxxxx-...-lb/inboundNatPools/yyyyyyyyyyy-pool)
azurerm_lb_rule.yyyyyyyyyyy-https: Refreshing state... (ID: /subscriptions/xxxxxxxxxxxxxxxxxxx-...i-lb/loadBalancingRules/HTTPSllllllllRule)
azurerm_lb_rule.yyyyyyyyyyy-http: Refreshing state... (ID: /subscriptions/xxxxxxxxxxxxxxxxxxx-...gi-lb/loadBalancingRules/HTTPllllllllRule)
azurerm_virtual_machine_scale_set.yyyyyyyyyyy: Refreshing state... (ID: /subscriptions/xxxxxxxxxxxxxxxxxxx-....Compute/virtualMachineScaleSets/llllllll)
Error refreshing state: 3 error(s) occurred:

* azurerm_virtual_machine_scale_set.yyyyyyyyyyy: 1 error(s) occurred:

* azurerm_virtual_machine_scale_set.yyyyyyyyyyy: azurerm_virtual_machine_scale_set.yyyyyyyyyyy: unexpected EOF
* azurerm_lb_rule.yyyyyyyyyyy-http: 1 error(s) occurred:

* azurerm_lb_rule.yyyyyyyyyyy-http: azurerm_lb_rule.yyyyyyyyyyy-http: unexpected EOF
* azurerm_lb_rule.yyyyyyyyyyy-https: 1 error(s) occurred:

* azurerm_lb_rule.yyyyyyyyyyy-https: azurerm_lb_rule.yyyyyyyyyyy-https: unexpected EOF
panic: runtime error: invalid memory address or nil pointer dereference
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1ed879f]
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4:
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: goroutine 158 [running]:
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.flattenAzureRmVirtualMachineScaleSetNetworkProfile(0xc4202f4120, 0x220101a, 0x17, 0x1fbe440)
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_virtual_machine_scale_set.go:903 +0x27f
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmVirtualMachineScaleSetRead(0xc420210700, 0x21220c0, 0xc4205a8000, 0x0, 0x2bea4c0)
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_virtual_machine_scale_set.go:733 +0xcc3
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc420516d20, 0xc42016a820, 0x21220c0, 0xc4205a8000, 0xc4202df8d0, 0x20c3f01, 0x80000000010)
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:321 +0x199
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc4204a7c00, 0xc42016a7d0, 0xc42016a820, 0x30d8458, 0x0, 0x0)
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:284 +0x9a
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc420141700, 0xc420342600, 0xc420342710, 0x0, 0x0)
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x4e
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: reflect.Value.call(0xc420053260, 0xc42016d7b8, 0x13, 0x21ebca6, 0x4, 0xc42029ff20, 0x3, 0x3, 0x0, 0x0, ...)
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/reflect/value.go:434 +0x906
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: reflect.Value.Call(0xc420053260, 0xc42016d7b8, 0x13, 0xc42080cf20, 0x3, 0x3, 0x0, 0x0, 0x0)
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: net/rpc.(*service).call(0xc42020acc0, 0xc42016b040, 0xc4201712a8, 0xc4203bd280, 0xc420141fa0, 0x1f9f180, 0xc420342600, 0x16, 0x1f9f1c0, 0xc420342710, ...)
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/net/rpc/server.go:381 +0x142
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: created by net/rpc.(*Server).ServeCodec
2017-10-31T10:04:58.319+0100 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/net/rpc/server.go:475 +0x36b
2017/10/31 10:04:58 [ERROR] root: eval: *terraform.EvalRefresh, err: azurerm_lb_rule.yyyyyyyyyyy-http: unexpected EOF
2017/10/31 10:04:58 [ERROR] root: eval: *terraform.EvalRefresh, err: azurerm_virtual_machine_scale_set.yyyyyyyyyyy: unexpected EOF
2017/10/31 10:04:58 [ERROR] root: eval: *terraform.EvalSequence, err: azurerm_virtual_machine_scale_set.yyyyyyyyyyy: unexpected EOF
2017/10/31 10:04:58 [ERROR] root: eval: *terraform.EvalRefresh, err: azurerm_lb_rule.yyyyyyyyyyy-https: unexpected EOF
2017/10/31 10:04:58 [TRACE] [walkRefresh] Exiting eval tree: azurerm_virtual_machine_scale_set.yyyyyyyyyyy
2017/10/31 10:04:58 [ERROR] root: eval: *terraform.EvalSequence, err: azurerm_lb_rule.yyyyyyyyyyy-http: unexpected EOF
2017/10/31 10:04:58 [TRACE] [walkRefresh] Exiting eval tree: azurerm_lb_rule.yyyyyyyyyyy-http
2017/10/31 10:04:58 [ERROR] root: eval: *terraform.EvalSequence, err: azurerm_lb_rule.yyyyyyyyyyy-https: unexpected EOF
2017/10/31 10:04:58 [TRACE] [walkRefresh] Exiting eval tree: azurerm_lb_rule.yyyyyyyyyyy-https
2017/10/31 10:04:58 [TRACE] dag/walk: upstream errored, not walking "provider.azurerm (close)"
2017-10-31T10:04:58.324+0100 [DEBUG] plugin: plugin process exited: path="/Users/XXXXXX/lb-llllllll/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v0.3.2_x4"
2017/10/31 10:04:58 [DEBUG] plugin: waiting for all plugin processes to complete...
2017-10-31T10:04:58.564+0100 [WARN ] plugin: error closing client during Kill: err="connection is shut down"
```

--- a/azurerm/resource_arm_virtual_machine_scale_set.go
+++ b/azurerm/resource_arm_virtual_machine_scale_set.go
@@ -898,9 +898,12 @@ func flattenAzureRmVirtualMachineScaleSetNetworkProfile(profile *compute.Virtual
 	result := make([]map[string]interface{}, 0, len(*networkConfigurations))
 	for _, netConfig := range *networkConfigurations {
 		s := map[string]interface{}{
-			"name":                   *netConfig.Name,
-			"primary":                *netConfig.VirtualMachineScaleSetNetworkConfigurationProperties.Primary,
-			"accelerated_networking": *netConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking,
+			"name":    *netConfig.Name,
+			"primary": *netConfig.VirtualMachineScaleSetNetworkConfigurationProperties.Primary,
+		}
+
+		if v := netConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking; v != nil {
+			s["accelerated_networking"] = *v
 		}
 
 		if v := netConfig.VirtualMachineScaleSetNetworkConfigurationProperties.NetworkSecurityGroup; v != nil {

